src/java.base/share/classes/java/lang/invoke/Invokers.java

```diff
@@ -122,6 +128,14 @@ private synchronized MethodHandle setCachedInvoker(int idx, final MethodHandle i
         return invokers[idx] = invoker;
     }
 
+    private MethodHandle cachedVHInvoker(int idx, VarHandle.AccessMode ak) {
```

Comment:
You can turn `int idx` parameter into `boolean isExact` and choose base index depending on its value.